src/main/groovy/org/shipkit/internal/gradle/contributors/ContributorsPlugin.java

```diff
@@ -16,7 +16,7 @@
  *     <li>fetchAllContributors - {@link AllContributorsFetcherTask}</li>
  * </ul>
  */
-public class ContributorsPlugin implements Plugin<Project> {
+public abstract class ContributorsPlugin implements Plugin<Project> {
```

Comment:
Can you explain in the description of the PR what is the motivation to create abstract classes? Abstract classes make the hierarchy complicated and usually it is better to replace the type hierarchy with composition. Plugin hierarchy is also harder to document. Can you try making the implementation use composition (e.g. one plugin applies another plugin)? Keeping the code simple! :)

We can also play along and see what happens - e.g. how your next PR demonstrates the need for plugin type hierarchies.